src/spec.rs

```diff
@@ -674,25 +524,234 @@ impl Zip64CentralDirectoryEnd {
             number_of_files,
             central_directory_size,
             central_directory_offset,
+            extensible_data_sector,
         } = self;
-        Zip64CDEBlock {
-            magic: Zip64CDEBlock::MAGIC,
-            /* currently unused */
-            record_size: 44,
-            version_made_by,
-            version_needed_to_extract,
-            disk_number,
-            disk_with_central_directory,
-            number_of_files_on_this_disk,
-            number_of_files,
-            central_directory_size,
-            central_directory_offset,
-        }
+
+        (
+            Zip64CDEBlock {
+                magic: Zip64CDEBlock::MAGIC,
+                record_size,
+                version_made_by,
+                version_needed_to_extract,
+                disk_number,
+                disk_with_central_directory,
+                number_of_files_on_this_disk,
+                number_of_files,
+                central_directory_size,
+                central_directory_offset,
+            },
+            extensible_data_sector,
+        )
     }
 
     pub fn write<T: Write>(self, writer: &mut T) -> ZipResult<()> {
-        self.block().write(writer)
+        let (block, comment) = self.block_and_comment();
+        block.write(writer)?;
+        writer.write_all(&comment)?;
+        Ok(())
+    }
+}
+
+pub(crate) struct CentralDirectoryEndInfo {
+    pub eocd: (Zip32CentralDirectoryEnd, u64),
```

Comment:
Split this into two fields for readability.

Reply:
I'll rather introduce `DataWithPosition<T>` to keep it as a single field, because the `eocd64` is an `Option` and having to match two `Option`s when they both must be either `Some` or `None` at the same time would be tedious.